src/KmlViewer/XInputHelper/XInputSceneController.cs

```diff
@@ -203,5 +205,45 @@ private static void OnControllerPropertyChanged(DependencyObject d, DependencyPr
             if (newCtrl != null)
                 newCtrl.SetSceneView(d as SceneView);
         }
+
+        private static bool CamerasAreEqual(Camera lhs, Camera rhs)
+        {
+            if (ReferenceEquals(lhs, rhs))
+            {
+                return true;
+            }
+
+            if (lhs == null && rhs == null)
```

Comment:
`ReferenceEquals` already checks whether _both_ are null.
Instead, this should check whether _one_ is null.  Checking here will avoid potential NullReferenceExceptions lower in the method.
```suggestion
            if (lhs == null || rhs == null)
```